sqlglot/dialects/dialect.py

```diff
@@ -441,6 +444,19 @@ def quote_identifier(self, expression: E, identify: bool = True) -> E:
 
         return expression
 
+    def parse_json_path(self, path: t.Optional[exp.Expression]) -> t.Optional[exp.Expression]:
```

Comment:
why is this done in the dialect and not just in the parser?

Reply:
Because this is currently used in `sqlglot.parser.parse_extract_json_with_path`, which is basically a helper to dry up `FUNCTIONS` entries. That lambda doesn't have access to the parser instance (*), so we'd have to move ~7 such entries in `FUNCTION_PARSERS` instead; iirc we want to avoid that.

Maybe the `parse` prefix in this method is a misnomer, because we are not really "parsing" anything, we're just constructing a new expression out of a given one. At least that's how I think about it, so it didn't seem that bad to have it in the dialect.

Thoughts?

Reply:
maybe to_json_path